django/thunderstore/api/cyberstorm/serializers/team.py

```diff
@@ -26,6 +26,17 @@ def get_avatar(self, obj) -> Optional[str]:
         return get_user_avatar_url(obj.user)
 
 
+class CyberstormTeamAddMemberRequestSerialiazer(serializers.Serializer):
+    user = serializers.CharField()
+    role = serializers.CharField()
+
```

Comment:
The request serializer is intended for validation of the request, meaning you should use more precise field definitions. `ModelChoiceField` for example can be used to limit the options to a specific known field value in a table (such as username). As for role, you just need to restrict it to the valid role choices (which exist as an enum).

Reply:
I guess I can just use `serializers.ModelSerializer` here

Reply:
![image](https://github.com/thunderstore-io/Thunderstore/assets/8225825/b5683374-f0d8-4db1-baee-1ba9ad0493f9)

`ModelSerializer` comes with some bells & whistles we _don't_ want to accidentally end up using, which is why it's been avoided for the most part (such as potentially "leaking" info about fields that should not be exposed, or supporting object CRUD by default even if you want it to be read only).

Notably the `ModelChoiceField` is an in-project custom field (`serializer_fields.py` somewhere) and I'm unsure if DRF even has an equivalent by default.

Reply:
You can think of the `ModelSerializer` class as a replacement for Django's forms more than a class explicitly used for (de)serialization.

Reply:
Though I'm not sure how the `team_member` in response should be done with `ModelChoiceField`.

Reply:
You don't need to validate the response so you don't need to use it there. You use it in the request serializer since that you want to validate

Reply:
Okiedokie, ducked around and found out with the ModelSerializer. (It tried to create teammembers when one was not found)

The ModelChoiceSeriliazer didn't atleast work out of box. I'll look at it tomorrow 👍🤔